src/main/java/com/google/devtools/build/lib/remote/GrpcRemoteExecutor.java

```diff
@@ -152,6 +160,9 @@ public ExecuteResponse executeRemotely(ExecuteRequest request)
                   Operation o = replies.next();
                   operation.set(o);
                   waitExecution.set(!operation.get().getDone());
+                  if (receiver != null) {
+                    receiver.onNextOperation(o);
```

Comment:
This is my biggest concern with the change: The service may send replies even when the action is still in the queue. I'm saying that because our service does, and I don't see a constraint on this in the documentation.

(All other comments are optional.)

Reply:
You are right. I was assuming the action is executing as long as we get a reply from service.

How about use `metadata.stage` to check execution status if the reply from server has [`metadata`](https://github.com/bazelbuild/bazel/blob/4b6ad34dbba15dacebfb6cbf76fa741649cdb007/third_party/remoteapis/build/bazel/remote/execution/v2/remote_execution.proto#L1219) field, and fallback to above strategy otherwise? Since `metadata` is not returned by some services (e.g. RBE), we can't know the accurate execution status in that case. But once we get a reply from service, we know the connection is up and the service should execute the action as soon as possible.

Reply:
That sounds like a fine plan!